File: PongSharp/src/Data/Angel.cs
using System.Numerics;

namespace PongSharp.Data;

internal struct Angel(int degrees)
{
    private const float DegInRad = 57.2958f;
    private const int MinValue = 0;
    private const int MaxValue = 360;

    public int Degrees
    {
        get => degrees;
        set
        {
            degrees = value switch
            {
                > MaxValue => value - value / MaxValue * MaxValue,
                < MinValue => value + (value / MaxValue + 1) * MaxValue,
                _ => value
            };
        }
    }

    public Vector2 ToVector2() => new(MathF.Cos(degrees / DegInRad), -MathF.Sin(degrees / DegInRad));
}

File: src/PongGame/Data/Angel.c
#include "Angel.h"
#include "Float2.h"
#include "MyMathTools.h"

const float DegInRad = 57.2958f;
const int MinValue = 0;
const int MaxValue = 360;

void Angel_Set(Angel* angel, int value)
{
    if(value > MaxValue)
        angel->_degrees = value - (value / MaxValue) * MaxValue;
    else if(value < MinValue)
        angel->_degrees = value + (value / MaxValue + 1) * MaxValue;
    else
        angel->_degrees = value;
    
    angel->Degrees = angel->_degrees;
}

Float2 Angel_ToFloat2(Angel* angel)
{
    Float2 r = {MyMath_Cos(angel->_degrees / DegInRad), -MyMath_Sin(angel->_degrees / DegInRad)};
    return r;
}
